Makefile

```diff
@@ -14,7 +14,7 @@ no_test: programs
 programs: lib mbedtls_test
 	$(MAKE) -C programs
 
-lib:
+lib: generated_files
```

Comment:
In principle, this is a good thing: it's good to ensure that the generated files are up-to-date. However this is likely to cause problems for users who work in an environment where they can't re-generate the files (no python, or not the required python modules, or no host C compiler), and where the timestamps on the generated files aren't newer than the primary sources (because the timestamps is in the random order in which the files were copied).

This is more or less went wrong in https://jenkins-mbedtls.oss.arm.com/blue/organizations/jenkins/mbed-tls-pr-head/detail/PR-7980-head/12/pipeline/582 : in this case, the build environment doesn't support `make generated_files` (although that's actually a bug in the makefile). Nothing would need to actually be rebuilt since the timestamps of generated files are newer than primary sources, but in this case, the failure happens before `make` can check this.

I think the main issue (if `make` runs ok but the code to generate the files might fail) can be resolved with [order-only prerequisites](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html). But it would have to be on `generated_files` in the sub-makefiles: an order-only prerequisite doesn't make sense when the prerequisite is phony.

This is a rather complicated makefile change which should be done in its own pull request. Do you really need it here?

Reply:
Agreed, not in scope for this PR (but also agreed it's potentially a nice improvement for a separate PR, if you have some spare background time for it).